Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Collections/GridObjectCollection.cs

```diff
@@ -0,0 +1,446 @@
+                    // Then use the packer function to shift them into place
+                    for (int i = 0; i < NodeList.Count; i++)
+                    {
+                        newPos = VectorExtensions.ScatterMapping(nodeGrid[i], Radius);
+                        Collider nodeCollider = NodeList[i].transform.GetComponentInChildren<Collider>();
+                        if (nodeCollider != null)
+                        {
+                            // Make the radius the largest of the object's dimensions to avoid overlap
+                            Bounds bounds = nodeCollider.bounds;
+                            NodeList[i].Radius = Mathf.Max(Mathf.Max(bounds.size.x, bounds.size.y), bounds.size.z) * 0.5f;
+                        }
+                        else
+                        {
+                            // Make the radius a default value
+                            // TODO move this into a public field ?
+                            NodeList[i].Radius = 1f;
+                        }
+                        NodeList[i].transform.localPosition = newPos;
+                        UpdateNodeFacing(NodeList[i], orientType, newPos);
+                    }
+
+                    // Iterate [x] times
+                    // TODO move center, iterations and padding into a public field
+                    for (int i = 0; i < 100; i++)
+                    {
+                        IterateScatterPacking(NodeList, Radius);
+                    }
+                    break;
+            }
+        }
+
+        private void ResolveGridLayout(Vector3[] grid, float offsetX, float offsetY, LayoutOrderTypeEnum order)
+        {
+            int cellCounter = 0;
+            float iMax;
+            float jMax;
+
+            if (order == LayoutOrderTypeEnum.RowThenColumn)
+            {
+                iMax = Rows;
+                jMax = columns;
+            }
+            else
+            {
+                iMax = columns;
+                jMax = Rows;
+            }
+
+            for (int i = 0; i < iMax; i++)
+            {
+                for (int j = 0; j < jMax; j++)
+                {
+                    if (cellCounter < NodeList.Count)
+                    {
+                        grid[cellCounter].Set(
+                            ((i * CellWidth) - offsetX + halfCell.x) + NodeList[cellCounter].Offset.x, (-(j * CellHeight) + offsetY - halfCell.y) + NodeList[cellCounter].Offset.y, 0.0f);
+                    }
+                    cellCounter++;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Update the facing of a node given the nodes new position for facing orign with node and orientation type
+        /// </summary>
+        /// <param name="node"></param>
+        /// <param name="orientType"></param>
+        /// <param name="newPos"></param>
+        private void UpdateNodeFacing(ObjectCollectionNode node, OrientationTypeEnum orientType, Vector3 newPos = default(Vector3))
```

Comment:
why not just vector3.zero?

Reply:
Dejavu :D 
https://github.com/Microsoft/MixedRealityToolkit-Unity/pull/2229#discussion_r192543633

Reply:
Seems as though orientType and newPos are unused in the method. We can just remove them altogether.